includes/Analytics/Reports/Orders/Stats/ScheduleListener.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace WeDevs\Dokan\Analytics\Reports\Orders\Stats;
+
+use WeDevs\Dokan\Contracts\Hookable;
+
+/**
+ * Class ScheduleListener
+ *
+ * Listens to WooCommerce schedule events and triggers Dokan order synchronization and deletion.
+ */
+class ScheduleListener implements Hookable {
+    /**
+     * ScheduleListener constructor.
+     * Registers the hooks on instantiation.
+     */
+    public function __construct() {
+        $this->register_hooks();
+    }
+
+    /**
+     * Register hooks for WooCommerce analytics order events.
+     *
+     * @return void
+     */
+    public function register_hooks(): void {
+        add_action( 'woocommerce_analytics_update_order_stats', [ $this, 'sync_dokan_order' ] );
+        add_action( 'woocommerce_before_delete_order', [ $this, 'delete_order' ] );
```

Comment:
Should we delete stats?

**Yes**
- Report will be mismatch with the `dokan_vendor_balance`.

**No**
- Report will not be synced as per WC stats data.